mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -34,6 +34,9 @@ sealed trait Matrix extends Serializable {
   /** Number of columns. */
   def numCols: Int
 
+  /** Flag that keeps track whether the matrix is transposed or not. False by default. */
+  private[linalg] var isTransposed = false
```

Comment:
We do modify it in place. In the transpose functions. We don't specify
isTransposed in the constructor.
On Jan 26, 2015 1:38 PM, "Xiangrui Meng" <notifications@github.com> wrote:

> In mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala
> <https://github.com/apache/spark/pull/4109#discussion_r23566423>:
>
> > @@ -34,6 +34,9 @@ sealed trait Matrix extends Serializable {
> >    /** Number of columns. */
> >    def numCols: Int
> >
> > +  /** Flag that keeps track whether the matrix is transposed or not. False by default. */
> > +  private[linalg] var isTransposed = false
>
> We should make this public because otherwise users won't be able to
> understand public members like values. This should be a val instead of var
> because we don't modify it in-place.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/4109/files#r23566423>.
>